### PR TITLE
ddl: refine fatal log when upgrade failed because of tidb_enable_dist_task

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

- `&*job` is equivalent to `job`, it should use `job.Clone()` to create another instance.
- The log of upgrade failed is not clear.
- There is a typo in log "task scheduler exist".

### What changed and how does it work?

Fix above three problems.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: pkg/ddl/table.go [hunk 1]
```diff
@@ -239,8 +239,8 @@ func onCreateTables(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, error) {
 	// We don't construct jobs for every table, but only tableInfo
 	// The following loop creates a stub job for every table
 	//
-	// &*job clones a stub job from the ActionCreateTables job
-	stubJob := &*job
+	// it clones a stub job from the ActionCreateTables job
+	stubJob := job.Clone()
```
Comment (D3Hunter) [inline pkg/ddl/table.go:243]: does `CtxVars` matters here? it's not cloned

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/ddl/table.go (hunk 1 above)
Comment (tangenta) [reply pkg/ddl/table.go:243]: `CtxVars` is only used by "reorg partition" and "rename tables", it is not used in "create tables".

File: pkg/session/bootstrap.go [hunk 2]
```diff
@@ -1348,7 +1348,8 @@ func checkDistTask(s sessiontypes.Session, ver int64) {
 		// Not set yet.
 		return
 	} else if req.GetRow(0).GetString(0) == variable.On {
-		logutil.BgLogger().Fatal("check dist task failed, tidb_enable_dist_task is enabled", zap.Error(err))
+		logutil.BgLogger().Fatal("running upgrade DDL is not supported when tidb_enable_dist_task is enabled, "+
```
Comment (tangenta) [inline pkg/session/bootstrap.go:1351]:
```suggestion
		logutil.BgLogger().Fatal("cannot upgrade when tidb_enable_dist_task is enabled, "+
```

**Pull request merged**